tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -1171,6 +1171,17 @@ void test_gh_5253() {
     g_regexTester.should_not_match("a", "()*");
 }
 
+void test_gh_5374() {
+    // GH-5374: <regex>: Back-references to unmatched capture groups
+    // should not match in POSIX basic regular expressions
+    for (syntax_option_type option : {basic, grep}) {
+        g_regexTester.should_not_match("", R"(\(.\)*\1)", option);
+        g_regexTester.should_match("", R"(\(.*\)\1)", option);
+        g_regexTester.should_not_match("bc", R"(\(a\)*b\1c)", option);
+        g_regexTester.should_match("bc", R"(\(a*\)b\1c)", option);
+    }
+}

```

Comment:
Hmm. The fact that you didn't have to edit any existing tests tells me that we didn't have any coverage for any of the grammars. Which means that we should add a test for ECMAScript, to verify that it does match. I will verify and push changes.